cmd/controller-manager/app/controller-manager.go

```diff
@@ -243,20 +243,33 @@ func getKubeFedConfig(opts *options.Options) *corev1b1.KubeFedConfig {
 	return fedConfig
 }
 
-func setDefaultKubeFedConfigScope(fedConfig *corev1b1.KubeFedConfig) {
-	if len(fedConfig.Spec.Scope) == 0 {
-		// TODO(sohankunkerkar) Remove when no longer necessary.
-		// This Environment variable is a temporary addition to support Red Hat's downstream testing efforts.
-		// Its continued existence should not be relied upon.
-		const defaultScopeEnv = "DEFAULT_KUBEFED_SCOPE"
-		defaultScope := os.Getenv(defaultScopeEnv)
-		if len(defaultScope) != 0 {
-			if defaultScope != string(apiextv1b1.ClusterScoped) && defaultScope != string(apiextv1b1.NamespaceScoped) {
-				klog.Fatalf("%s must be Cluster or Namespaced; got %q", defaultScopeEnv, defaultScope)
-			}
+func setDefaultKubeFedConfigScope(fedConfig *corev1b1.KubeFedConfig) bool {
+	// TODO(sohankunkerkar) Remove when no longer necessary.
+	// This Environment variable is a temporary addition to support Red Hat's downstream testing efforts.
+	// Its continued existence should not be relied upon.
+	const defaultScopeEnv = "DEFAULT_KUBEFED_SCOPE"
+	defaultScope := os.Getenv(defaultScopeEnv)
+	if len(defaultScope) != 0 {
+		if defaultScope != string(apiextv1b1.ClusterScoped) && defaultScope != string(apiextv1b1.NamespaceScoped) {
+			klog.Fatalf("%s must be one of %s or %s; got %q", defaultScopeEnv,
+				string(apiextv1b1.ClusterScoped), string(apiextv1b1.NamespaceScoped), defaultScope)
+			return false
+		}
+
+		if len(fedConfig.Spec.Scope) == 0 {
 			fedConfig.Spec.Scope = apiextv1b1.ResourceScope(defaultScope)
+			klog.Infof("Setting the scope of KubeFedConfig spec to %s", defaultScope)
+			return true
+		} else {
+			if fedConfig.Spec.Scope != apiextv1b1.ResourceScope(defaultScope) {
+				klog.Infof("Setting the scope of KubeFedConfig spec from %s to %s",
+					string(fedConfig.Spec.Scope), defaultScope)
+				fedConfig.Spec.Scope = apiextv1b1.ResourceScope(defaultScope)
```

Comment:
Apologies for not bringing this up earlier, but we should only log within the if/else and the identical statements can be performed afterwards.